### PR TITLE
Removing deprecated hash variable which was used to cache the transaction

### PR DESCRIPTION
**Pull request by mohitsaxenaknoldus**
## Fixed Issue(s)

#2937 

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

**Comment by mohitsaxenaknoldus** [comment]
> ## PR description
> ## Fixed Issue(s)
#2937 
> ## Changelog
> * [x]  I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

**Comment by macfarla** [comment]
Looks ok to me @mohitsaxenaknoldus 
However the unit test step is failing because the plugin API hash needs updating as a result of this change:
```
* What went wrong:
Execution failed for task ':plugin-api:checkAPIChanges'.
> For the Plugin APIs the checksum of the project did not match what was expected.
  
  If this is a deliberate change where you have thought through backwards compatibility,
  then update "Expected" for "Calculated" in the appropriate build.gradle as the knownHash for this task.
  Expected   : /M23BAhwqtMpjhcoHDMJvbHyhJ4z39z2jJMWUGOBO14=
  Calculated : iVl4LwulrgbVqYsahA4YvgrMqc22OxBk43hFsoYPA5w=
```

**Comment by mohitsaxenaknoldus** [comment]
> Looks ok to me @mohitsaxenaknoldus However the unit test step is failing because the plugin API hash needs updating as a result of this change:
> 
> ```
> * What went wrong:
> Execution failed for task ':plugin-api:checkAPIChanges'.
> > For the Plugin APIs the checksum of the project did not match what was expected.
>   
>   If this is a deliberate change where you have thought through backwards compatibility,
>   then update "Expected" for "Calculated" in the appropriate build.gradle as the knownHash for this task.
>   Expected   : /M23BAhwqtMpjhcoHDMJvbHyhJ4z39z2jJMWUGOBO14=
>   Calculated : iVl4LwulrgbVqYsahA4YvgrMqc22OxBk43hFsoYPA5w=
> ```

Oh ok. How can I fix that?

**Comment by macfarla** [comment]
> > Looks ok to me @mohitsaxenaknoldus However the unit test step is failing because the plugin API hash needs updating as a result of this change:
> > ```
> > * What went wrong:
> > Execution failed for task ':plugin-api:checkAPIChanges'.
> > > For the Plugin APIs the checksum of the project did not match what was expected.
> >   
> >   If this is a deliberate change where you have thought through backwards compatibility,
> >   then update "Expected" for "Calculated" in the appropriate build.gradle as the knownHash for this task.
> >   Expected   : /M23BAhwqtMpjhcoHDMJvbHyhJ4z39z2jJMWUGOBO14=
> >   Calculated : iVl4LwulrgbVqYsahA4YvgrMqc22OxBk43hFsoYPA5w=
> > ```
> 
> Oh ok. How can I fix that?
You have to do exactly what it says in the message above - update the knownHash in plugin-api/build.gradle

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Changelog
 
+## 21.10.4
+
+### Additions and Improvements
+- Removed deprecated hash variable `protected volatile Hash hash;` which was used for private transactions [#3110](https://github.com/hyperledger/besu/pull/3110)
```
Comment (macfarla) [inline CHANGELOG.md:6]: this should be under 21.10.3 since we haven't released that version yet (latest is 21.10.2)

**Comment by mohitsaxenaknoldus** [comment]
@macfarla Fixed the chandelog, please check now. Sorry for the delays!

**Comment by macfarla** [comment]
@mohitsaxenaknoldus the changelog entry for this should be moved under the topmost release in the changelog ie 22.1.0-RC2
Also the hash needs updating in 
Build file '/home/circleci/project/plugin-api/build.gradle' line: 53

  If this is a deliberate change where you have thought through backwards compatibility,
  then update "Expected" for "Calculated" in the appropriate build.gradle as the knownHash for this task.
  Expected   : DhISJBKoARZyoiT387UIA7QfcEwyOcIYm17f/fLsJRM=
  Calculated : pDLQUs2Gl9hFV1ecKSacbbfwovrjMf/IBPiY1zkOZlI=

**Comment by mohitsaxenaknoldus** [comment]
> @mohitsaxenaknoldus the changelog entry for this should be moved under the topmost release in the changelog ie 22.1.0-RC2 Also the hash needs updating in Build file '/home/circleci/project/plugin-api/build.gradle' line: 53
> 
> If this is a deliberate change where you have thought through backwards compatibility, then update "Expected" for "Calculated" in the appropriate build.gradle as the knownHash for this task. Expected : DhISJBKoARZyoiT387UIA7QfcEwyOcIYm17f/fLsJRM= Calculated : pDLQUs2Gl9hFV1ecKSacbbfwovrjMf/IBPiY1zkOZlI=

Done.
Also, here it is showing DCO ok but on CI DCO is failing, why?

**Comment by macfarla** [comment]
> Also, here it is showing DCO ok but on CI DCO is failing, why?
There's a DCO issue on another branch. We are working on it, should be fixed shortly.

**Comment by mohitsaxenaknoldus** [comment]
> > Also, here it is showing DCO ok but on CI DCO is failing, why?
> > There's a DCO issue on another branch. We are working on it, should be fixed shortly.

Oh ok, seems to be passing now.

File: plugin-api/build.gradle [hunk 2]
```diff
@@ -54,8 +54,8 @@ class FileStateChecker extends DefaultTask {
 
 If this is a deliberate change where you have thought through backwards compatibility,
 then update "Expected" for "Calculated" in the appropriate build.gradle as the knownHash for this task.
-Expected   : ${this.knownHash}
-Calculated : ${currentHash}
+Expected   : DhISJBKoARZyoiT387UIA7QfcEwyOcIYm17f/fLsJRM=
+Calculated : pDLQUs2Gl9hFV1ecKSacbbfwovrjMf/IBPiY1zkOZlI=
```
Comment (macfarla) [inline plugin-api/build.gradle:58]: we don't want to update this bit!

this line is what needs changing:
  knownHash = 'QPxUj/HJM7DZkbUq538/S2f3MmwPOqaPX/BBTkKznQk='

**Review by macfarla — CHANGES_REQUESTED** [review]

**Comment by mohitsaxenaknoldus** [comment]
Let's see if it works now @macfarla

**Comment by mohitsaxenaknoldus** [comment]
Hey @macfarla Is this good to merge now?

**Comment by macfarla** [comment]
Somehow the hash got overwritten again in a "merge from main" commit. Fingers crossed I've fixed it now! Thanks for your patience with this one @mohitsaxenaknoldus !

**Review by taccatisid — APPROVED** [review]

**Review by macfarla — APPROVED** [review]

**Comment by macfarla** [comment]
Thanks again for your patience with getting this change merged in @mohitsaxenaknoldus - hopefully your next contribution doesn't hit so many bumps in the road!!

**Pull request merged**